doc/design/model_market.md

```diff
@@ -0,0 +1,119 @@
+# Model Market
+
+The model market is a web site where model developers can publish and share their models and analysts can find some useful models to finish the analysis work. The model market can be deployed anywhere like on the cloud or on-premise with some configurations.
+
+In the [model zoo design](model_zoo.md), we described how model developers develop, publish and share custom models on SQLFlow, and how analysts can make use of the shared model with SQLFlow.
+
+Here, we'll describe how to build the model market model developers and analysts.
+
+## Overview
+
+The model market is designed to:
+
+1. Manage user login and logout.
+1. View published images available for the current user.
+1. View published trained models available for the current user.
+1. Publish (or remove) model definition images.
+1. Publish (or remove) trained models.
+1. Share (or remove share) model definition images to other users.
+1. Share (or remove share) trained models to other users.
+
+<p align="center">
+<img src="figures/model_market_overview.png">
```

Comment:
@tonyyang-svail Could we embed GraphViz plots in Markdown design docs on GitHub? It would be easier to do code review if the figure is in text format.

Reply:
You can find examples [here](https://github.com/TLmaK0/gravizo) to embed GraphViz plots.